mmv1/templates/terraform/examples/service_attachment_cross_region_ilb.tf.tmpl

```diff
@@ -0,0 +1,77 @@
+resource "google_compute_service_attachment" "{{$.PrimaryResourceId}}" {
+  name                  = "{{index $.Vars "name"}}"
+  region                = "us-central1"
+  description           = "A service attachment configured with Terraform"
+  connection_preference = "ACCEPT_AUTOMATIC"
+  enable_proxy_protocol = false
+  nat_subnets           = [google_compute_subnetwork.subnetwork_psc.id]
+  target_service        = google_compute_global_forwarding_rule.forwarding_rule.id
+}
+
+resource "google_compute_global_forwarding_rule" "forwarding_rule" {
+  name                  = "{{index $.Vars "name"}}"
+  target                = google_compute_target_http_proxy.http_proxy.id
+  network               = google_compute_network.network.id
+  subnetwork            = google_compute_subnetwork.subnetwork.id
+  port_range            = "80"
+  load_balancing_scheme = "INTERNAL_MANAGED"
+
+  depends_on = [google_compute_subnetwork.subnetwork_proxy]
+}
+
+resource "google_compute_target_http_proxy" "http_proxy" {
+  name        = "{{index $.Vars "name"}}"
+  description = "a description"
+  url_map     = google_compute_url_map.url_map.id
+}
+
+resource "google_compute_url_map" "url_map" {
+  name            = "{{index $.Vars "name"}}"
+  description     = "Url map."
+  default_service = google_compute_backend_service.backend_service.id
+}
+
+resource "google_compute_backend_service" "backend_service" {
+  name                  = "{{index $.Vars "name"}}"
+  load_balancing_scheme = "INTERNAL_MANAGED"
+  health_checks         = [google_compute_health_check.health_check.id]
+}
+
+resource "google_compute_health_check" "health_check" {
+  name               = "{{index $.Vars "name"}}"
+  check_interval_sec = 1
+  timeout_sec        = 1
+
+  tcp_health_check {
+    port = "80"
+  }
+}
+
+resource "google_compute_subnetwork" "subnetwork_psc" {
+  name          = "psc-{{index $.Vars "name"}}"
```

Comment:
The name in the generated provider is `psc-tf-test-tf-test%{random_suffix}`, starting with `psc` https://github.com/modular-magician/terraform-provider-google-beta/compare/b4758b2cf0010a4c6e8426d29c74eb0c0fa97e25..f9b0ce33dbafd0aced577ac26bbf63fb8b696546#diff-9019649d6ee1a84552b947ba345402c59ef3a12112ba50dfcb46a85fdc48e420R562

Instead, the name should start with `tf-test`, which is added automatically to `{{index $.Vars "name"}}`. The leaked resources with `tf-test` prefix could be sweeped by the sweeper.

A new variable could be used, similar to `name` in ServiceAttachment.yaml file